### PR TITLE
Remove checkpoints from custom plots

### PR DESCRIPTION
**Pull request by mattseddon**
#  #3585 <- this

This PR removes checkpoints from custom plots. Once this is merged into #3585 we should be able to merge that PR. I will update that PR with a final description and make note of any outstanding items.

### Demo

https://user-images.githubusercontent.com/37993418/229422864-2ec9160b-ae48-4b2c-8db8-8de47220e976.mov

1 of the 2 users who had trend plots (migrating between versions):

https://user-images.githubusercontent.com/37993418/229423991-ae0df7a5-0353-4f9f-9838-9122e5433141.mov

### CODE REVIEW

File: webview/src/plots/components/customPlots/customPlotsSlice.ts [hunk 1]
```diff
@@ -58,7 +55,7 @@ export const customPlotsSlice = createSlice({
       if (!action.payload) {
         return customPlotsInitialState
       }
-      const { plots, colors, ...statePayload } = action.payload
```
Comment (mattseddon) [inline webview/src/plots/components/customPlots/customPlotsSlice.ts:61]: [Q] @julieg18 why were we sending colors? I can't find any references to it.

**Review by sroy3 — APPROVED** [review]

File: webview/src/plots/components/customPlots/customPlotsSlice.ts (hunk 1 above)
Comment (julieg18) [reply webview/src/plots/components/customPlots/customPlotsSlice.ts:61]: Leftover from #3575! They can be deleted :)

File: extension/src/plots/model/custom.ts [hunk 2]
```diff
@@ -35,23 +19,17 @@ export const getFullValuePath = (type: string, columnPath: string) =>
 export const cleanupOldOrderValue = (
   value: { metric: string; param: string } | CustomPlotsOrderValue
 ): CustomPlotsOrderValue => ({
-  // previous column paths have the "TYPE:" prefix
   metric: removeColumnTypeFromPath(value.metric, ColumnType.METRICS),
-  param: removeColumnTypeFromPath(value.param, ColumnType.PARAMS),
-  // previous values didn't have a type
-  type: (value as CustomPlotsOrderValue).type || CustomPlotType.METRIC_VS_PARAM
```
Comment (julieg18) [inline extension/src/plots/model/custom.ts:42]: Are we sure about wanting to delete the `type` in workspace state values? If we want to add a different type of custom plot in the future, we're going to have to adjust our state again, adding the type back. Might be easier to just keep it even if it's not being used in the code currently 🤔

**Review by julieg18 — APPROVED** [review]
Great work!

File: extension/src/plots/model/custom.ts (hunk 2 above)
Comment (mattseddon) [reply extension/src/plots/model/custom.ts:42]: YAGNI

**Pull request merged**